bin/citrea/tests/bitcoin_e2e/light_client_test.rs

```diff
@@ -143,3 +149,317 @@ async fn test_light_client_proving() -> Result<()> {
+        let mut cfg = SequencerConfig {
+            min_soft_confirmations_per_commitment: 20,
+            da_update_interval_ms: 500,
+            ..Default::default()
+        };
+
+        cfg.mempool_conf.max_account_slots = 400;
+        cfg
+    }
+
+    fn batch_prover_config() -> BatchProverConfig {
+        BatchProverConfig {
+            enable_recovery: false,
+            ..Default::default()
+        }
+    }
+
+    fn light_client_prover_config() -> LightClientProverConfig {
+        LightClientProverConfig {
+            enable_recovery: false,
+            ..Default::default()
+        }
+    }
+
+    async fn run_test(&mut self, f: &mut TestFramework) -> Result<()> {
+        let da = f.bitcoin_nodes.get(0).unwrap();
+        let sequencer = f.sequencer.as_ref().unwrap();
+        let batch_prover = f.batch_prover.as_ref().unwrap();
+        let light_client_prover = f.light_client_prover.as_ref().unwrap();
+        let full_node = f.full_node.as_ref().unwrap();
+
+        let min_soft_confirmations_per_commitment =
+            sequencer.min_soft_confirmations_per_commitment();
+
+        let rpc_bind_host = sequencer.config().rpc_bind_host();
+        let rpc_bind_port = sequencer.config().rpc_bind_port();
+
+        let socket_addr = SocketAddr::new(rpc_bind_host.parse().unwrap(), rpc_bind_port);
+
+        let seq_test_client = make_test_client(socket_addr).await.unwrap();
+
+        // publish min_soft_confirmations_per_commitment confirmations
+        for _ in 0..4 * min_soft_confirmations_per_commitment {
+            // sequencer.client.http_client().eth_send_raw_transaction(data)
+            for _ in 0..30 {
+                let address = Address::random();
+                let _pending = seq_test_client
+                    .send_eth(address, None, None, None, 1u128)
+                    .await
+                    .unwrap();
+            }
+            sequencer.client.send_publish_batch_request().await?;
+        }
+        sequencer
+            .wait_for_l2_height(4 * min_soft_confirmations_per_commitment, None)
+            .await?;
+
+        // Wait for commitment txs to be submitted to DA
+        da.wait_mempool_len(8, Some(TEN_MINS)).await.unwrap();
+
+        // Finalize the DA block which contains the commitment txs
+        da.generate(FINALITY_DEPTH, None).await.unwrap();
+
+        let commitment_l1_height = da.get_finalized_height().await.unwrap();
+
+        // Wait for batch prover to generate proofs for commitments
+        batch_prover
+            .wait_for_l1_height(commitment_l1_height, Some(Duration::from_secs(1200)))
+            .await
+            .unwrap();
+
+        // Assert that commitments are queryable
+        let commitments = batch_prover
+            .client
+            .http_client()
+            .get_sequencer_commitments_on_slot_by_number(commitment_l1_height)
+            .await
+            .unwrap()
+            .unwrap();
+        assert_eq!(commitments.len(), 4);
```

Comment:
can you fix this thing on other tests as well. remmeber our realization about commit / reveal